controllers/application/egress_service_entry.go

```diff
@@ -22,138 +13,35 @@ func (r *ApplicationReconciler) reconcileEgressServiceEntry(ctx context.Context,
 	controllerName := "EgressServiceEntry"
 	r.SetControllerProgressing(ctx, application, controllerName)
 
-	if application.Spec.AccessPolicy != nil {
-		for _, rule := range application.Spec.AccessPolicy.Outbound.External {
-			name := fmt.Sprintf("%s-egress-%x", application.Name, util.GenerateHashFromName(rule.Host))
-
-			serviceEntry := networkingv1beta1.ServiceEntry{ObjectMeta: metav1.ObjectMeta{Namespace: application.Namespace, Name: name}}
-			_, err := ctrlutil.CreateOrPatch(ctx, r.GetClient(), &serviceEntry, func() error {
-				// Set application as owner of the service entry
-				err := ctrlutil.SetControllerReference(application, &serviceEntry, r.GetScheme())
-				if err != nil {
-					r.SetControllerError(ctx, application, controllerName, err)
-					return err
-				}
-				r.SetLabelsFromApplication(ctx, &serviceEntry, *application)
-				util.SetCommonAnnotations(&serviceEntry)
-
-				resolution, adresses, endpoints := getIpData(rule.Ip)
+	serviceEntries := istio.GetServiceEntries(application.Spec.AccessPolicy, application)
 
-				serviceEntry.Spec = networkingv1beta1api.ServiceEntry{
-					// Avoid leaking service entry to other namespaces
-					ExportTo:   []string{".", "istio-system", "istio-gateways"},
-					Hosts:      []string{rule.Host},
-					Resolution: resolution,
-					Addresses:  adresses,
-					Endpoints:  endpoints,
-					Ports:      r.getPorts(rule.Ports, rule.Ip, *application),
-				}
-
-				return nil
-			})
+	for _, serviceEntry := range serviceEntries {
+		_, err := ctrlutil.CreateOrPatch(ctx, r.GetClient(), &serviceEntry, func() error {
+			// Set application as owner of the service entry
+			err := ctrlutil.SetControllerReference(application, &serviceEntry, r.GetScheme())
 			if err != nil {
 				r.SetControllerError(ctx, application, controllerName, err)
-				return reconcile.Result{}, err
+				return err
 			}
-		}
-	}
-
-	// Clear out unused service entries
-	serviceEntries := networkingv1beta1.ServiceEntryList{}
-	err := r.GetClient().List(ctx, &serviceEntries, client.InNamespace(application.Namespace))
-	if err != nil {
-		r.SetControllerError(ctx, application, controllerName, err)
-		return reconcile.Result{}, err
-	}
+			r.SetLabelsFromApplication(ctx, &serviceEntry, *application)
+			util.SetCommonAnnotations(&serviceEntry)
 
-	for _, serviceEntry := range serviceEntries.Items {
-		// Skip unrelated service entries
-		if !isEgressServiceEntry(serviceEntry) {
```

Comment:
I see this check is removed, I assume this is also because it is only iterating over owned resources?

Reply:
Aye!

Reply:
Or actually, no. In the deletion phase we list all the service entries in the namespace. However the check after, as defined in the `DeleteUnusedEgresses` function in the ReconcilerBase, checks for ownership either way. Result should be the same